lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Hai Cao, <t-haicao@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_cdnendpoint_facts
+
+version_added: "2.8"
+
+short_description: Get Azure CDN endpoint facts
+
+description:
+    - Get facts for a specific Azure CDN endpoint or all Azure CDN endpoints.
+
+options:
+    resource_group:
+        description:
+            - Name of resource group where this CDN profile belongs to.
+        required: true
+    profile_name:
+        description:
+            - Name of CDN profile.
+        required: true
+    name:
+        description:
+            - Limit results to a specific Azure CDN endpoint.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Hai Cao (@caohai)"
+    - "Yunge zhu (@yungezz)"
+'''
+
+EXAMPLES = '''
+  - name: Get facts for all endpoints in CDN profile
+    azure_rm_cdnendpoint_facts:
+      resource_group: myResourceGroup
+      profile_name: myCDNProfile
+
+  - name: Get facts of specific CDN endpoint
+    azure_rm_cdnendpoint_facts:
+      resource_group: myResourceGroup
+      profile_name: myCDNProfile
+      name: myEndpoint1
+'''
+
+RETURN = '''
+cdnendpoints:
+    description: List of Azure CDN endpoints.
+    returned: always
+    type: complex
+    contains:
+        resource_group:
+            description:
+                - Name of a resource group where the Azure CDN endpoint exists.
+            returned: always
+            type: str
+            sample: myResourceGroup
+        name:
+            description:
+                - Name of the Azure CDN endpoint.
+            returned: always
+            type: str
+            sample: myEndpoint
+        profile_name:
+            description:
+                - Name of the Azure CDN profile that this endpoint is attached to.
+            returned: always
+            type: str
+            sample: myProfile
+        state:
```

Comment:
maybe we can drop state. it was my idea in the beginning, but i think it's not really necessary now as always the same.